pkg/products/threescale/reconciler.go

```diff
@@ -3113,7 +3115,21 @@ func (r *Reconciler) reconcileRatelimitingTo3scaleComponents(ctx context.Context
 		ratelimit.RateLimitClusterName,
 		getRatelimitServicePort(ratelimitServiceCR),
 	)
-	ratelimitClusterResource.Http2ProtocolOptions = &envoycorev3.Http2ProtocolOptions{}
+
+	serial, err := anypb.New(&envoyextentionv3.HttpProtocolOptions{
+		UpstreamProtocolOptions: &envoyextentionv3.HttpProtocolOptions_ExplicitHttpConfig_{
+			ExplicitHttpConfig: &envoyextentionv3.HttpProtocolOptions_ExplicitHttpConfig{
+				ProtocolConfig: &envoyextentionv3.HttpProtocolOptions_ExplicitHttpConfig_Http2ProtocolOptions{},
+			},
+		},
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	ratelimitClusterResource.TypedExtensionProtocolOptions = map[string]*any.Any{
+		"envoy.extensions.upstreams.http.v3.HttpProtocolOptions": serial,
+	}
```

Comment:
Deprecated in favour of https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/upstreams/http/v3/http_protocol_options.proto